framework/auth/views.py

```diff
@@ -409,7 +420,7 @@ def external_login_confirm_email_get(auth, uid, token):
             mimetype='html',
             user=user
         )
-        service_url = service_url + '?new=true'
+        service_url += '&{}'.format(urllib.urlencode({'new': 'true'}))
```

Comment:
`service_url = request.url`, where `request.url` can only be the confirmation link with `next` parameter when this part of the code is reached. Will add a test for this.